### PR TITLE
[AC-110: Feature] Verify user email when register

### PR DESCRIPTION
**Pull request by miaozhaod**
### Provide description as to what you have done in this pr.
- Update `register` function to allow user receive verification email
- Create `verifyUserEmail` function to update UserStatus into `verified`

### Provide screenshots or videos that the function is working.
https://user-images.githubusercontent.com/89491470/180720514-767402a2-a6ba-473f-b477-e6a0a864a7f8.mov

### Provide screenshots of no warning, error in console.
image

#### That the feature you have worked on has its...

~~- [ ] relevant information updated in `README.MD`~~
~~- [ ] swagger doc created/updated so that other developers can read the related documentation.~~
- [x] logging included/updated so that we can debug errors when the app is live in the UAT environment.
~~- [ ] unit tests created/updated so that we can integrate this new release and sleep without worries at night/day.~~

#### When you finished, you will want to make sure you have...

- [x] No merge conflict with `develop` branch.
- [x] Check PR title is correct, (i.e `[AC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there's hard coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/resource/auth/auth.resolver.ts [hunk 1]
```diff
@@ -34,6 +35,11 @@ export class AuthResolver {
     return this.authService.register(_registerDetail);
   }
 
+  @Mutation(() => Boolean)
+  verifyUserEmail(@Args('detail') verifyEmailInput: VerifyEmailInput) {
+    return this.authService.verifyUserEmail(verifyEmailInput);
```
Comment (NorrisWu0) [inline src/resource/auth/auth.resolver.ts:43]: Need to add logging here

File: src/resource/auth/auth.service.ts [hunk 2]
```diff
@@ -81,6 +83,21 @@ export class AuthService {
     };
   }
 
+  async verifyUserEmail({ id }: VerifyEmailInput) {
+    const user = await this.userService.findById(id);
+    if (!user) {
+      this.logger.error(`User with id ${id} does not exist`);
+      return;
```
Comment (NorrisWu0) [inline src/resource/auth/auth.service.ts]: `this.logger.error` will throw an exception when used, so we don't need to `return` here

File: src/resource/user/user.resolver.ts [hunk 3]
```diff
@@ -63,4 +64,13 @@ export class UserResolver {
     );
     return this.userService.updateProfile(data);
   }
+
+  @Mutation(() => Boolean)
+  @UseGuards(JwtAuthGuard)
+  verifyAccountSendEmail(@Args('id') id: VerifyAccountSendEmailInput) {
+    this.logger.log(
+      `Receiving request to update profile of user with id ${id}...`,
```
Comment (NorrisWu0) [inline src/resource/user/user.resolver.ts]: ?

File: src/resource/user/user.service.ts [hunk 4]
```diff
@@ -52,6 +58,32 @@ export class UserService {
...
+      email,
+      subject,
+      emailContent,
+    });
+
+    this.logger.log(
+      'axiosService.sendEmail of verify user email result ...',
+      result,
+    );
+    if (!result) return false;
```
Comment (NorrisWu0) [inline src/resource/user/user.service.ts]: I think it's better to log a separate `logger.error` for this?

**Review by NorrisWu0 — CHANGES_REQUESTED** [review]

File: src/resource/auth/auth.resolver.ts (hunk 1 above)
Comment (miaozhaod) [reply src/resource/auth/auth.resolver.ts:43]: updated by adding log

File: src/resource/auth/auth.service.ts (hunk 2 above)
Comment (miaozhaod) [reply src/resource/auth/auth.service.ts]: updated with throw error

File: src/resource/user/user.resolver.ts (hunk 3 above)
Comment (miaozhaod) [reply src/resource/user/user.resolver.ts]: updated with proper text

File: src/resource/user/user.service.ts (hunk 4 above)
Comment (miaozhaod) [reply src/resource/user/user.service.ts]: updated with log err

**Review by NorrisWu0 — APPROVED** [review]

**Pull request merged**